### DISCUSSION TITLE
Bad CPU type in executable - Adobe Animate CC 2019

### DISCUSSION THREAD

**Discussion by affy69** [airsdk/Adobe-Runtime-Support#1823] [Q&A]
Hi 

I'm no developer, wonder if you can assist

I am using Adobe Animate CC 2019 and trying to Publishing AIR3.0 for IOS. 

My Operating system isCatalina 10:15:7

I have a project that I want to publish straight to my iPhone 12 Max for testing. Device is selected it in the device selection and all the Certificates and Provisioning profiles have been attached.

But I am getting this error:

/var/folders/0z/hs2nvdh565n9snnfyvcf038r0000gn/T/cmd1252528715338691008.tmp: line 2: /Applications/Adobe Animate CC 2019/AIR30.0/lib/aot/bin/compile-abc/compile-abc: Bad CPU type in executable

Thanks in advance.
<img width="561" alt="1" src="https://user-images.githubusercontent.com/103079997/161867633-989a65ab-a0a9-40fa-b1e0-2217d6d04773.png">
<img width="538" alt="2" src="https://user-images.githubusercontent.com/103079997/161867635-961824da-b4ce-4d0e-b6e4-3a2f442a724e.png">
<img width="476" alt="3" src="https://user-images.githubusercontent.com/103079997/161867638-5e86e661-efcc-493c-80e3-acbdaf7d737b.png">

**Comment by ajwfrost** [comment]
Yes, this "compile-abc" tool is a 32-bit binary so won't run on Catalina. You would need to use Mojave if you want to use an older AIR SDK like this.

If you take a recent AIR SDK from https://airsdk.harman.com then the tool will work out you're running on Catalina, and will not call the 32-bit compile-abc tool; it means your IPA file will also only contain a 64-bit binary though so will only work on 64-bit versions of iOS.

thanks
